src/profile/webid.ts

```diff
@@ -94,3 +95,60 @@ export async function getProfileAll<
     altProfileAll,
   };
 }
+
+/**
+ * Discover the Pods an agent advertises for in their profile resources. Both the
+ * agent's WebID and alternative profiles are fetched. Note that this function will
+ * only return URLs of Pods linked to using the `pim:storage`, i.e. a triple
+ * looking like <myWebid, pim:storage, myPodUrl> should appear in the profile
+ * resources.
+ *
+ * @param webId The WebID of the agent whose Pods should be discovered
+ * @param options Optional parameter
+ * - `options.fetch`: An alternative `fetch` function to make the HTTP request,
+ *    compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns a Promise resolving to an array containing the URLs of all the Pods
+ * linked from the agent's profile resource using the `pim:storage` predicate.
+ */
+export async function getPodUrlAll(
+  webId: WebId,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): Promise<UrlString[]> {
+  const profiles = await getProfileAll(webId, options);
+  return getPodUrlAllFrom(profiles, webId);
+}
+
+/**
+ * Discover the Pods advertised for in the provided profile resources. Note that
+ * this function will only return URLs of Pods linked to using the `pim:storage`
+ * predicate, i.e. a triple looking like <myWebid, pim:storage, myPodUrl>
+ * should appear in the profile resources.
+ *
+ * @param profiles The profile resources in which the Pods should be discovered
+ * @param webId The WebID of the agent whose Pods should be discovered
+ * @returns An array containing the URLs of all the Pods linked from the agent's
+ * profile resource using the `pim:storage` predicate.
+ */
+export function getPodUrlAllFrom<
+  T extends SolidDataset & WithServerResourceInfo
+>(profiles: ProfileAll<T>, webId: WebId): UrlString[] {
```

Comment:
Did we not establish that Generics should at least appear twice?
https://github.com/inrupt/solid-client-js/pull/1356
https://microsoft.github.io/TypeScript-New-Handbook/everything/#type-parameters-should-appear-twice
```suggestion
export function getPodUrlAllFrom(
  profiles: ProfileAll<SolidDataset & WithServerResourceInfo>,
  webId: WebId
): UrlString[] {
```

Reply:
That's a very good point ^^

Reply:
Fixed in d4272cf5b710e0c1191383affa6e42fada7349e0